src/datagrid/DataGrid.tsx

```diff
@@ -312,6 +293,76 @@ export class DataGrid extends React.PureComponent<DataGridProps, DataGridState>
         }
     }
 
+    // funtion to initialize datagrid state
+    initializeDataGridState = (): DataGridState => {
+        const {isLoading, itemText} = this.props;
+        const rows = this.initializeRowData();
+        const columns = this.initializeColumnData();
+        const state: DataGridState = {
+            isLoading: isLoading || false,
+            selectAll: this.isAllRowsSelected(rows),
+            allColumns: [...columns],
+            allRows: [...rows],
+            itemText: itemText || DEFAULT_ITEM_TEXT,
+            pagination: this.initializePaginationData(),
+        };
+        return state;
+    };
+
+    // Function to initialize rows data in state
+    initializeRowData = (): DataGridRow[] => {
+        const {rows} = this.props;
+        let updatedRows: DataGridRow[] = [];
+        if (rows && rows.length) {
+            updatedRows = this.updateRowIDs(rows);
+            updatedRows.forEach(function(row) {
+                const rowSelectionIsDisabled = row.disableRowSelection !== undefined ? row.disableRowSelection : false;
+                row.isSelected = !rowSelectionIsDisabled && row.isSelected !== undefined ? row.isSelected : false;
+            });
+        }
+        return updatedRows;
+    };
+
+    // Function to initialize columns data in state
+    initializeColumnData = (): DataGridColumn[] => {
+        const {columns} = this.props;
+        let updatedColumns: DataGridColumn[] = [];
+        if (columns && columns.length) {
+            updatedColumns = this.updateColumnIDs(this.setColumnVisibility(columns));
+            updatedColumns.forEach(col => {
+                col.width = col.width ? col.width : DEFAULT_COLUMN_WIDTH;
+            });
+        }
+        return updatedColumns;
+    };
+
+    // Initialize state of grid with pagination
+    private initializePaginationData() {
+        const {pagination} = this.props;
+        if (pagination) {
+            const {currentPage, pageSize, totalItems, compactFooter, pageSizes} = pagination;
+            const currentPageNumber: number = currentPage || DEFAULT_CURRENT_PAGE_NUMBER;
+            const datagridPageSize: number = pageSize || DEFAULT_PAGE_SIZE;
+            const totalItemsInDatagrid: number = totalItems || DEFAULT_TOTAL_ITEMS;
+
+            const firstItem: number = this.getFirstItemIndex(currentPageNumber, datagridPageSize);
+            const lastItem: number = this.getLastItemIndex(datagridPageSize, totalItemsInDatagrid, firstItem);
+
+            const paginationState: DataGridPaginationState = {
+                currentPage: currentPageNumber,
+                pageSize: datagridPageSize,
+                totalItems: totalItemsInDatagrid,
+                pageSizes: pageSizes,
+                compactFooter: compactFooter || false,
```

Comment:
Might be old code, but why is this called CompactFooter and why not Footer ?
do we have Footer(big) as well ?

Reply:
@maheshkulkarni4 This is for compact footer which we show when objectStoreDetails page is open

![image](https://user-images.githubusercontent.com/51195071/96410951-567b7500-1205-11eb-83f1-aee480399947.png)